c_src/main.c

```diff
@@ -86,23 +86,16 @@ erlfdb_future_cb(FDBFuture* fdb_future, void* data)
     cancelled = future->cancelled;
     enif_mutex_unlock(future->cancel_lock);
 
-    enif_mutex_lock(future->msg_lock);
```

Comment:
The lock seems useless—it saves only from concurrent `erlfdb_future_cb` invocations, which the fdb driver should prevent.
But if it somehow happens, then we will anyway run into problems calling `enif_release_resource(future)` twice.